docs/java/features/odata/generate-client.md

```diff
@@ -0,0 +1,270 @@
+        <!-- Please use the latest version here-->
+        <version>3.18.0</version>
+        <executions>
+            <execution>
+                <id>generate-consumption</id>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>generate</goal>
+                </goals>
+                <configuration>
+                    <inputDirectory>${project.basedir}/edmx</inputDirectory>
+                    <outputDirectory>${project.build.directory}/vdm</outputDirectory>
+                    <deleteOutputDirectory>true</deleteOutputDirectory>
+                    <packageName>com.mycompany.vdm</packageName>
+                    <defaultBasePath>odata/v4/</defaultBasePath>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    ```
+
+</TabItem>
+</Tabs>
+
+2. Adapt the `<inputDirectory>` to point to the place of your service definition.
+
+1. In case the target folder should be automatically added as a source folder by maven you can leverage the build helper plugin:
+    ```XML
+    <plugin>
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <version>3.0.0</version>
+        <executions>
+            <execution>
+                <phase>generate-sources</phase>
+                <goals>
+                    <goal>add-source</goal>
+                </goals>
+                <configuration>
+                    <sources>
+                        <source>${project.build.directory}/vdm</source>
+                    </sources>
+                </configuration>
+            </execution>
+        </executions>
+    </plugin>
+    ```
+
+Now maven will run the generator within the `process-sources` phase which is executed before compile.
+
+The following parameters allow to configure the generator further:
+
+|   Parameter Value |   Explanation |
+|-------------------|---------------|
+|`<inputDirectory>`|Location of the metadata file.
+|`<outputDirectory>`|Location of the output directory for generated sources.
+|`<deleteOutputDirectory>`|Target directory is deleted before every execution of the generator
+|`<packageName>`|Package name for the generated sources
+|`<defaultBasePath>`|Base path of the exposed API
+
+:::note
+Please note that if you use the generator for services other than SAP S/4HANA services, you need to add the parameter `defaultBasePath` to the configuration section, which should provide the base path to the exposed API (e.g odata/v4/).
+:::
+
+### Using the CLI
+
+<Tabs defaultValue="v4" values={[
+{ label: 'OData V2', value: 'v2', },
+{ label: 'OData V4', value: 'v4', }]}>
+<TabItem value="v4">
+
+1. Download the command line interface (CLI) of the generator from [internal nexus](http://nexus.wdf.sap.corp:8081/nexus/) (search for `odata-v4-generator-cli`), rename it to `odata-generator-cli.jar` and put it in a folder of your choice.
```

Comment:
We can't put internal documentation here!?

Reply:
Good catch! It's not internal, I changed the instructions to download from maven central.